packages/apps/plugins/plugin-explorer/src/components/Tree/Tree.tsx

```diff
@@ -0,0 +1,108 @@
+//
+// Copyright 2023 DXOS.org
+//
+
+import React, { useEffect, useMemo, useState } from 'react';
+import { useResizeDetector } from 'react-resize-detector';
+
+import { type Space } from '@dxos/client/echo';
+import { createSvgContext, SVG, SVGContextProvider } from '@dxos/gem-core';
+
+import { HierarchicalEdgeBundling, RadialTree, TidyTree } from './layout';
+import { mapGraphToTreeData, type TreeNode } from './types';
+import { SpaceGraphModel } from '../Graph';
+
+// TODO(burdon): Create dge bundling graph using d3.hierarchy.
+// https://observablehq.com/@d3/hierarchical-edge-bundling?intent=fork
+
+type Renderer = (svg: SVGSVGElement, data: any, options: any) => void;
+
+export type LayoutType = 'tidy' | 'radial' | 'edge';
+
+// TODO(burdon): Normalize API and styling.
+
+export type TreeLayoutSlots = {
+  node?: string;
+  path?: string;
+  text?: string;
+};
+
+export type TreeOptions = {
+  label: (d: any) => string;
+
+  slots?: TreeLayoutSlots;
+  radius?: number;
+
+  width: number;
+  height: number;
+  margin?: number;
+
+  padding?: number;
+  // Radius of nodes.
+  r?: number;
+};
+
+export const defaultTreeLayoutSlots: TreeLayoutSlots = {
+  node: 'fill-blue-600',
+  path: 'fill-none stroke-blue-400 stroke-[0.5px]',
+  text: 'stroke-[0.5px] stroke-neutral-700 text-xs', // TODO(burdon): Create box instead of halo.
+};
+
+const renderers = new Map<LayoutType, Renderer>([
+  ['tidy', TidyTree],
+  ['radial', RadialTree],
+  ['edge', HierarchicalEdgeBundling],
+]);
+
+export type TreeComponentProps<N = unknown> = {
+  space: Space;
```

Comment:
This could be better supplied by a context rather than drilling it down through props.